qrbill/bill.py

```diff
@@ -51,52 +51,58 @@
 
 
 class Address:
-    def __init__(self, *, name='', street='', house_num='', pcode=None, city=None, country=None):
-        if not (1 < len(name or '') < 70):
+    def __init__(self, *, name=None, street=None, house_num=None, pcode=None, city=None, country=None):
+        self.name = name or ''
```

Comment:
If, for example, street is not specified on the command line, then we get a None value passed to __init__ anyway from the caller, due to argparse not having a value for the argument. Keeping that as None can cause problems in the QR-code.

This could also be used as a library in another program, without using scripts/qrbill. Where does the data from such another program come from? If there is, let's say no street and house number in the customer DB of that program, will it pass None or an empty string? We don't know.

So, basically, in the processing code we have to make sure we convert input to sensible values for each of the passed parameters anyway. In that case I think it is better to be consistent in what the defaults in __init__ are (we could also agree on '') and have some kind of common processing-path to follow in the code for each parameter. That makes it easier to understand and easier to extend in the future, should that ever be necessary.